sdk/monitor/monitor-query/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,304 @@
+service if they run for more than 3 minutes. For such scenarios, the query APIs on `LogsQueryClient`, provide options to
+configure the timeout on the server. The server timeout can be extended up to 10 minutes.
+
+You may see an error as follows:
+
+```text
+Code: GatewayTimeout
+Message: Gateway timeout
+Inner error: {
+    "code": "GatewayTimeout",
+    "message": "Unable to unzip response"
+}
+```
+
+The following code shows a sample on how to set the server timeout to 10 minutes. Note that by setting this server
+timeout, the Azure Monitor Query library will automatically also extend the client timeout to wait for 10 minutes for
+the server to respond. You don't need to configure your HTTP client to extend the response timeout as shown in the
+previous section.
+
+```ts
+import { DefaultAzureCredential } from "@azure/identity";
+import { LogsQueryClient } from "@azure/monitor-query";
+
+const credential = new DefaultAzureCredential();
+const logsQueryClient = new LogsQueryClient(credential);
+const result = await logsQueryClient.queryWorkspace(
+        monitorWorkspaceId,
+        kustoQuery,
+        { duration: Durations.oneHour },
+        {serverTimeoutInSeconds: 600}
+        );
+```
+
+### Troubleshooting partially successful logs query requests
+
+By default, if the execution of a Kusto query resulted in a partially successful response, the Azure Monitor Query
+client library will return the result of type `LogsQueryPartialResult` with the `status` field of `result` object set to `PartialFailure` to indicate to the user that the query was not fully successful. In case of multiple queries, when the results from all queries aren't successful, the `status` field of `result` object may be set to `Failure`.
+
+In case of single query, there can be only two possibilities for the value of `status` field of `result` object - `Success` or `PartialFailure`. You can access the details of the partially successful results by the following code snippet -
+
+```ts
+const result = await logsQueryClient.queryWorkspace(${azureLogAnalyticsWorkspaceId}, ${kustoQuery}, {
+    duration: ${duration}
+  });
+
+  if (result.status === LogsQueryResultStatus.Success) {
+    const tablesFromResult: LogsTable[] = result.tables;
+
+    if (tablesFromResult.length === 0) {
+      console.log(`No results for query '${kustoQuery}'`);
+      return;
+    }
+    console.log(`This query has returned table(s) - `);
+    processTables(tablesFromResult);
+  } else {
+    console.log(`Error processing the query '${kustoQuery}' - ${result.partialError}`);
+    if (result.partialTables.length > 0) {
+      console.log(`This query has also returned partial data in the following table(s) - `);
+      processTables(result.partialTables);
+    }
+  }
+```
+More details on the hierarchy of the response for single query can be found [here](https://github.com/KarishmaGhiya/azure-sdk-for-js/blob/monitor-query-TS/sdk/monitor/monitor-query/README.md#handle-logs-query-response)
```

Comment:
Use the Azure org's link instead of your forked repo.


```suggestion
More details on the hierarchy of the response for single query can be found [here](https://github.com/Azure/azure-sdk-for-js/blob/monitor-query-TS/sdk/monitor/monitor-query/README.md#handle-logs-query-response)
```